opencog/nlp/relex2logic/rule-helpers.scm

```diff
@@ -414,30 +414,35 @@
 )
 ; -----------------------------------------------------------------------
 ; Adjective and adverb rules
+; (advmod is also called for adverbials, such as some prepositional phrases)
+;
+; Prepositional adverbial example: "The man sang with a powerful voice."
+; Prepositional adjectival example: "The man with a powerful voice sang."
+;
 ; -----------------------------------------------------------------------
 (define (amod-rule concept instance adj adj_instance)
 	(list (InheritanceLink  (ConceptNode adj_instance df-node-stv) (ConceptNode adj df-node-stv) df-link-stv)
-	(InheritanceLink  (ConceptNode instance df-node-stv) (ConceptNode adj_instance df-node-stv) df-link-stv)
 	(InheritanceLink  (ConceptNode instance df-node-stv) (ConceptNode concept df-node-stv) df-link-stv)
-	)
+	(InheritanceLink  (ConceptNode instance df-node-stv) (ConceptNode adj_instance df-node-stv) df-link-stv))
 )
 
 (define (advmod-rule verb instance adv adv_instance)
 	(list (InheritanceLink  (ConceptNode adv_instance df-node-stv) (ConceptNode adv df-node-stv) df-link-stv)
-	(InheritanceLink  (SatisfyingSetLink (PredicateNode instance df-node-stv) df-link-stv) (ConceptNode adv_instance df-node-stv) df-link-stv)
 	(ImplicationLink  (PredicateNode instance df-node-stv) (PredicateNode verb df-node-stv) df-link-stv)
+	(InheritanceLink  (SatisfyingSetLink (PredicateNode instance df-node-stv) df-link-stv) (ConceptNode adv_instance df-node-stv) df-link-stv)
 	)
 )
-
-;(define (prepadvmod-rule verb_concept verb_instance noun_concept noun_instance prep_concept prep_instance)
-;	(list 
-;		(InheritanceLink (ConceptNode noun_instance df-node-stv) (ConceptNode noun_concept df-node-stv) df-link-stv)
-;		(InheritanceLink (ConceptNode prep_instance df-node-stv) (ConceptNode prep_concept df-node-stv) df-link-stv)
-;		
-;		(ImplicationLink (PredicateNode verb_instance df-node-stv) (PredicateNode verb_concept df-node-stv) df-link-stv)
-;		(InheritanceLink (SatisfyingSetLink (PredicateNode instance df-node-stv) df-link-stv) (ConceptNode adv_instance df-node-stv) df-link-stv)
-;	)
-;)
+;-----------------------------------------------------------------------
+; prepositional phrase rule
+;-----------------------------------------------------------------------
+(define (pp-rule prep_concept prep_instance noun_concept noun_instance)
+	(list (ImplicationLink (PredicateNode prep_instance df-node-stv) (PredicateNode prep_concept df-node-stv) df-link-stv)
+	(InheritanceLink (ConceptNode noun_instance df-node-stv) (ConceptNode noun_concept df-node-stv) df-link-stv)
+	(EvaluationLink df-link-stv
+		(PredicateNode prep_instance df-node-stv)
+		(ListLink (ConceptNode noun_instance df-node-stv))
+	))
+)
```

Comment:
Yes, in general, a lot of the changes I have made make the construction of the logical form more modular.  This seems to me to make sense, because for example, a prepositional phrase may serve the role of an adjective (an adjectival), as in "Look at the book on the table," an adverb (an adverbial), as in "Sit on the table, please," or an intensional complement, as in "the book is on the table."  And likewise, many things other than prepositional phrases can play each of these roles, so rather than having a separate rule for each kind of phrase in each role, it makes more sense to just have a rule for each kind of phrase, and a rule for each role, and then put those together -- fewer rules, greater flexibility.

As for SVO, need to look at that again.  Treating "The book is on the table" as SVO is not technically corect, from the point of view of linguistics, but I may have instantiated it that way in order to adapt to the limitations of relex output at the time i did those revisions. We can probably do better.